src/context.rs

```diff
@@ -260,63 +177,26 @@ impl<A: Actor> Context<A> {
 
     /// Handle a message and immediate notifications, returning whether to exit from the manage loop
     /// or not.
-    async fn tick(
-        &mut self,
-        msg: Either<BroadcastMessage<A>, AddressMessage<A>>,
-        actor: &mut A,
-    ) -> ContinueManageLoop {
+    async fn tick(&mut self, msg: ActorMessage<A>, actor: &mut A) -> ContinueManageLoop {
         match msg {
-            Either::Left(BroadcastMessage::Message(msg)) => msg.handle(actor, self).await,
-            Either::Left(BroadcastMessage::Shutdown) => {
+            ActorMessage::BroadcastMessage(msg) => msg.handle(actor, self).await,
+            ActorMessage::Shutdown => {
                 self.running = RunningState::Stopped;
                 return ContinueManageLoop::ExitImmediately;
             }
-            Either::Right(AddressMessage::Message(msg)) => {
-                msg.handle(actor, self).await;
-            }
-            Either::Right(AddressMessage::LastAddress) => {
-                if self.ref_counter.strong_count() == 0 {
-                    self.stop_all();
-                    self.running = RunningState::Stopped;
-                    return ContinueManageLoop::ExitImmediately;
-                }
-            }
+            ActorMessage::StolenMessage(msg) => msg.handle(actor, self).await,
```

Comment:
I figured that that is what you named it after. Also agreed that unicast is not actually better.

What we are doing here might also not strictly be work-stealing though? Not sure if this is being too pedantic but in the Wikipedia article you linked, work stealing is defined as each processor having its own queue and them stealing work from others as they are idle.

In our case, we have a single heap of messages to be processed and actors consume them as they become ready to handle a message. Maybe `NextMessge` (or just `Next` because `Message` is duplicated with the enum name)? Or `PoppedMessage`?

Reply:
Fair points. Perhaps if we rename `Broadcast` too we can say:

- `AddressMessage::ToAllActors`
- `AddressMessage::ToOneActor`

Reply:
That could be a nice solution!

Reply:
I've renamed it now